Hudl.Mjolnir/Command/Attribute/InvocationCommand.cs

```diff
@@ -38,8 +46,8 @@ protected override Task<TResult> ExecuteAsync(CancellationToken cancellationToke
             // Do we have a CancellationToken property? If so, where is it? Will be -1 if not found.
             var cancellationTokenIndex = new List<ParameterInfo>(_invocation.Method.GetParameters()).FindLastIndex(IsCancellationToken);
 
-            // If we have one that doesn't already have a value, lets give it ours.
-            if (cancellationTokenIndex >= 0 && IsReplaceableToken(_invocation.Arguments[cancellationTokenIndex]))
+            // If we have one that doesn't already have a value, lets give it ours. Provided that we haven't got a fully disabled timeouts override, since the token should be None in that case
+            if (cancellationTokenIndex >= 0 && IsReplaceableToken(_invocation.Arguments[cancellationTokenIndex]) && !TimeoutsIgnored)
```

Comment:
I was hoping to keep all knowledge of the TimeoutsIgnored in the base class, but we need to make sure  InvocationCommand doesn't add a timeout token in the case when we want it ignored.